src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -0,0 +1,332 @@
+    // console.log('update', type, props, this.children);
+    // need to manually update text nodes
+    this.type = type;
+    this.props = props;
+  }
+
+  toJSON() {
+    // eslint-disable ignore the children
+    const {children, ...props} = this.props;
+    // eslint-enable
+    const json: ReactTestRendererJSON = {
+      type: this.type,
+      props: props,
+      children: null,
+    };
+
+    Object.defineProperty(json, '$$typeof', {value: Symbol.for('react.test.json')});
+    if (typeof children === 'string') {
+      json.children = [children];
+    } else {
+      var childrenJSON = [];
+      this.children.forEach((child) => {
+        if (typeof child.toJSON === 'function') {
+          childrenJSON.push(child.toJSON());
+        } else if (typeof child.text !== 'undefined') {
+          childrenJSON.push();
+        }
+      });
+      json.children = childrenJSON.length ? childrenJSON : null;
+    }
+    return json;
+  }
+}
+
+type Container = TestContainer;
+type Props = Object;
+type Instance = TestComponent;
+type TextInstance = {
+  text: string | number,
+  id: number,
+  rootContainerInstance: Container,
+  toJSON(): string | number,
+};
+
+var TestRenderer = ReactFiberReconciler({
+  getRootHostContext() {
+    return emptyObject;
+  },
+
+  getChildHostContext() {
+    return emptyObject;
+  },
+
+  prepareForCommit() : void {
+    // noop
+  },
+
+  resetAfterCommit() : void {
+    // noop
+  },
+
+  createInstance(
+    type : string,
+    props : Props,
+    rootContainerInstance : Container,
+    hostContext : Object,
+    internalInstanceHandle : Object,
+  ) : Instance {
+    return new TestComponent(type, props, rootContainerInstance);
+  },
+
+  appendInitialChild(parentInstance : Instance, child : Instance | TextInstance) : void {
+    parentInstance.appendChild(child);
+  },
+
+  finalizeInitialChildren(
+    testElement : Instance,
+    type : string,
+    props : Props,
+    rootContainerInstance : Container,
+  ) : boolean {
+    // console.log('finalizeInitialChildren');
+    // setInitialProperties(testElement, type, props, rootContainerInstance);
+    return false;
+  },
+
+  prepareUpdate(
+    testElement : Instance,
+    type : string,
+    oldProps : Props,
+    newProps : Props,
+    hostContext : Object,
+  ) : boolean {
+    return true;
+  },
+
+  commitUpdate(
+    instance : Instance,
+    type : string,
+    oldProps : Props,
+    newProps : Props,
+    rootContainerInstance : Container,
+    internalInstanceHandle : Object,
+  ) : void {
+    instance.update(type, newProps);
+  },
+
+  commitMount(
+    instance : Instance,
+    type : string,
+    newProps : Props,
+    rootContainerInstance : Object,
+    internalInstanceHandle : Object
+  ) : void {
+    // Noop
+  },
+
+  shouldSetTextContent(props : Props) : boolean {
+    return (
+      typeof props.children === 'string' ||
+      typeof props.children === 'number'
+    );
+  },
+
+  resetTextContent(testElement : Instance) : void {},
```

Comment:
Nit: Could you please use a consistent style for empty methods? Either like this, or with `// Noop` comment as long as it's the same style everywhere.